saleor/graphql/checkout/resolvers.py

```diff
@@ -33,30 +36,35 @@ def resolve_checkout(info, token, id):
     validate_one_of_args_is_in_query("id", id, "token", token)
 
     if id:
-        _, token = from_global_id_or_error(id, only_type="Checkout")
-    checkout = (
-        models.Checkout.objects.using(get_database_connection_name(info.context))
-        .filter(token=token)
-        .first()
-    )
-    if checkout is None:
-        return None
-    # always return checkout for active channel
-    if checkout.channel.is_active:
-        return SyncWebhookControlContext(node=checkout, allow_sync_webhooks=True)
-
-    # resolve checkout for staff or app
-    if requester := get_user_or_app_from_context(info.context):
-        has_manage_checkout = requester.has_perm(CheckoutPermissions.MANAGE_CHECKOUTS)
-        has_impersonate_user = requester.has_perm(AccountPermissions.IMPERSONATE_USER)
-        has_handle_payments = requester.has_perm(PaymentPermissions.HANDLE_PAYMENTS)
-        if has_manage_checkout or has_impersonate_user or has_handle_payments:
+        _, token = from_global_id_or_error(id, only_type="Checkout", raise_error=True)
+        token = UUID(token)
+
+    def with_checkout(checkout):
+        if checkout is None:
+            return None
+        # always return checkout for active channel
+        if checkout.channel.is_active:
```

Comment:
Shouldn't we also fetch the `channel` with the use of dataloader? In such case we are fetching the channel separately for each checkout